horovod/_keras/__init__.py

```diff
@@ -167,32 +185,83 @@ def allreduce_grads():
                             # `IndexedSlices`.
                             return [tf.identity(g) for g in averaged_gradients]
 
+            def trim_last_curly_brace(s):
+                if s[-1] != "}":
+                    raise AssertionError(f"Expected last character in \"{s}\" to be \"}}\", but got \"{s[-1]}\"")
+                return s[:-1]
+
+            def profile_start():
+                if self._profile_frequency == 0:
+                    return tf.no_op()
+                return tf.cond(tf.equal(self.profile_counter, self._profile_frequency - 1), log_comm_start, tf.no_op)
+
+            def log_comm_start():
+                profile_base_info = trim_last_curly_brace(get_profile_info("communication", "B"))
```

Comment:
Might be useful to explain this going the chrome formatting (hence the "B" and "E").